examples/systemcalls/systemcalls.c

```diff
@@ -79,21 +116,46 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
     {
         command[i] = va_arg(args, char *);
     }
+    va_end(args);
     command[count] = NULL;
     // this line is to avoid a compile warning before your implementation is complete
     // and may be removed
     command[count] = command[count];
 
+    pid_t pid = fork();
+    if (pid == -1) {
+        perror("fork");
+        return false;
+    } 
+    
+    // The child process
+    if (pid == 0) {
+        const int fd = creat(outputfile, 0644);
```

Comment:
@jackcenter think about how you could further avoid duplication with the function above.